src/main/java/com/gmail/nossr50/placeholders/PapiExpansion.java

```diff
@@ -49,105 +44,41 @@ public String getRequiredPlugin() {
     @Override
     @Nullable
     public String onPlaceholderRequest(final Player player, @NotNull final String params) {
-        String token;
-        String data = null;
-        int dataPosition = params.indexOf(":");
-
-        if (dataPosition != -1) {
-            token = params.substring(0, dataPosition);
-            data = params.substring(dataPosition + 1);
-        } else {
-            token = params;
-        }
-
-        Placeholder placeholder = placeholders.get(token);
-
-        if (placeholder != null) {
-            return placeholder.process(player, data);
-        } else {
-            return null;
-        }
-    }
-
-    public Integer getSkillLevel(PrimarySkillType skill, Player player) {
-        final McMMOPlayer user = UserManager.getPlayer(player);
-        if (user == null) return null;
-        return user.getSkillLevel(skill);
-    }
-
-    public Integer getExpNeeded(PrimarySkillType skill, Player player) {
-        final McMMOPlayer user = UserManager.getPlayer(player);
-        if (user == null) return null;
-        return user.getXpToLevel(skill);
-    }
-
-    public Integer getExp(PrimarySkillType skill, Player player) {
-        final McMMOPlayer user = UserManager.getPlayer(player);
-        if (user == null) return null;
-
-        return user.getSkillXpLevel(skill);
-    }
-
-
-    public Integer getExpRemaining(PrimarySkillType skill, Player player) {
-        final McMMOPlayer user = UserManager.getPlayer(player);
-        if (user == null) return null;
-        int current = user.getSkillXpLevel(skill);
-        int needed = user.getXpToLevel(skill);
-
-        return needed - current;
-    }
-
-    public Integer getRank(PrimarySkillType skill, Player player) {
+      
+      // Non player-specific placeholders
+      if (params.equalsIgnoreCase("is_xp_event_active")) {
+        return mcMMO.p.isXPEventEnabled() ? PlaceholderAPIPlugin.booleanTrue() : PlaceholderAPIPlugin.booleanFalse();
+      }else if (params.equalsIgnoreCase("xprate")) {
+        return String.valueOf(ExperienceConfig.getInstance().getExperienceGainsGlobalMultiplier());
+      }else if (params.equalsIgnoreCase("power_level_cap")) {
+        return mcMMO.p.getGeneralConfig().getPowerLevelCap()+"";
+      }
+      
+      final McMMOPlayer user = UserManager.getPlayer(player);
+      if (user == null) return null;
+      
+      if (params.startsWith("level_")) {
+        PrimarySkillType skill = PrimarySkillType.valueOf(params.substring(6).toUpperCase());
```

Comment:
you could use make use of constants to avoid these hard coded magic numbers
public static final String SKILL_LEVEL = "level_";

// later on
PrimarySkillType skill = PrimarySkillType.valueOf(params.substring(SKILL_LEVEL.length()).toUpperCase());

Reply:
Would you like this done for those only needing substring or all?
Eg. `"in_party"`

Reply:
for all would make sense to me